src/input.rs

```diff
@@ -40,12 +40,12 @@ impl<Manager> Input<Manager> {
         }
     }
 
-    /// Returns a list of the currently connected controllers without allocating
-    pub fn get_connected_controllers_slice(&self, mut controllers: impl AsMut<[InputHandle_t]>) {
+    /// Returns a list of the currently connected controllers without allocating, and the count
+    pub fn get_connected_controllers_slice(&self, mut controllers: impl AsMut<[InputHandle_t]>) -> usize {
         let handles = controllers.as_mut();
-        assert!(handles.len() <= sys::STEAM_INPUT_MAX_COUNT as usize);
+        assert!(handles.len() >= sys::STEAM_INPUT_MAX_COUNT as usize);
```

Comment:
the length shouldn't be less than or equal STEAM_INPUT_MAX_COUNT?

Reply:
from https://partner.steamgames.com/doc/api/isteaminput#GetConnectedControllers
> This must point to a [STEAM_INPUT_MAX_COUNT](https://partner.steamgames.com/doc/api/ISteamInput#STEAM_INPUT_MAX_COUNT) sized array of InputHandle_t.

combined with the fact that it doesn't take any length parameter indicates that it's going to think it has STEAM_INPUT_MAX_COUNT values available no matter what you pass to it, which means that you either want an == relationship or a >= relationship. If we want to switch it to ==, we could also do that by taking a reference to an array of the correct size.

Reply:
The example shown here also has the >= instead of <=

https://github.com/Noxime/steamworks-rs/pull/96#discussion_r860741989

Reply:
what if my game only supports 2 controllers? can't I send a "controllers" of size 2?

Reply:
There is no way to tell steam input that you only support 2 controllers, therefore it will assume you can handle an array of STEAM_INPUT_MAX_COUNT controllers, and will happly write out of range if you have 3 controllers.